core/src/modules/locks.ts

```diff
@@ -41,3 +41,35 @@ export async function waitForLock(
 
   return { releaseLock, attempts };
 }
+
+export type getLockResponse = {
+  releaseLock: Function;
+  isLocked: boolean;
+  lockedBy: string;
+};
+
+export async function getLock(
+  key: string,
+  requestId: string = uuid.v4(),
+  ttl = LOCK_DURATION_MS,
+  lockType?: string
+): Promise<getLockResponse> {
+  const client = api.redis.clients.client;
+  const lockKey = `grouparoo:lock:${
+    lockType.length > 1 && `${lockType}:`
+  }${key}`;
+
+  const set = await client.setnx(lockKey, requestId);
+
+  await client.expire(lockKey, Math.ceil(ttl / 1000));
+
+  const isLocked = !set;
+
+  async function releaseLock() {
+    await client.del(lockKey);
+  }
```

Comment:
Similar to the above, perhaps if you didn't create the lock, you shouldn't have access to the releaseLock method

Reply:
Yeah, maybe we call this `getLockIfAvailable` or something. It would return `null` if it's already locked and the `releaseLock` if we were able to get it.

Reply:
I think that makes sense.  During the unlocking in this particular instance, it doesn't much matter who created the lock.  The locks are just stored in an array and processed one-by-one after export.  But in terms of overall design, I think it makes sense to only give the `releaseLock()` out once.